models/migrations/v103.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"crypto/md5"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"code.gitea.io/gitea/modules/setting"
+
+	"xorm.io/xorm"
+)
+
+func renameExistingUserAvatarName(x *xorm.Engine) error {
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	type User struct {
+		ID     int64 `xorm:"pk autoincr"`
+		Avatar string
+	}
+
+	users := make([]User, 0)
+	if err := sess.Find(&users); err != nil {
+		return err
+	}
+
+	deleteList := make(map[string]struct{})
+	sessionOpCount := 0
+	for _, user := range users {
+		oldAvatar := user.Avatar
+		newAvatar := fmt.Sprintf("%x", md5.Sum([]byte(fmt.Sprintf("%d-%s", user.ID, user.Avatar))))
+		if _, err := os.Stat(filepath.Join(setting.AvatarUploadPath, oldAvatar)); err != nil {
+			continue
+		}
+
+		fr, err := os.Open(filepath.Join(setting.AvatarUploadPath, oldAvatar))
+		if err != nil {
+			_ = commitSession(sess)
+			return fmt.Errorf("os.Open: %v", err)
+		}
+		defer fr.Close()
```

Comment:
No problem.
Another thing: If we always have to migrate the avatars for all the users due to some failure, do we really need to commit the session with previous changes before returning the error ?
https://github.com/go-gitea/gitea/blob/8baca3992e6ce8cae9355952ddead870001b62c7/models/migrations/v103.go#L52-L57

Reply:
@masudur-rahman That's what I thought before you pointed me out that you couldn't delete the avatars until finishing migrating them all. Now it doesn't seem to add much value. But it's already in the code, so removing it is a matter of cleaning up the code if you want to. In a not-so-unrelated matter, I think it would be useful to print the user's `lower_name` as part of the error messages. But it's not strictly required.

Reply:
Done.
Everything's should be okay now. @guillep2k, can you please recheck now ?